avm/res/network/express-route-port/main.bicep

```diff
@@ -0,0 +1,207 @@
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.network-expressourteport.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource expressRoutePort_lock 'Microsoft.Authorization/locks@2020-05-01' = if (!empty(lock ?? {}) && lock.?kind != 'None') {
+  name: lock.?name ?? 'lock-${name}'
+  properties: {
+    level: lock.?kind ?? ''
+    notes: lock.?kind == 'CanNotDelete'
+      ? 'Cannot delete resource or child resources.'
+      : 'Cannot delete or modify the resource or child resources.'
+  }
+  scope: expressRoutePort
+}
+
+resource expressRoutePort 'Microsoft.Network/ExpressRoutePorts@2024-05-01' = {
+  name: name
+  location: location
+  properties: {
+    bandwidthInGbps: bandwidthInGbps
+    billingType: billingType
+    encapsulation: encapsulation
+    links: !empty(links) ? links : []
+    peeringLocation: peeringLocation
+  }
+}
+
+resource expressRouteCircuit_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(expressRoutePort.id, roleAssignment.principalId, roleAssignment.roleDefinitionId)
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: expressRoutePort
+  }
+]
+
+@description('The resource ID of the ExpressRoute Gateway.')
+output resourceId string = expressRoutePort.id
+
+@description('The resource group of the ExpressRoute Gateway was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The name of the ExpressRoute Gateway.')
+output name string = expressRoutePort.name
+
+@description('The location the resource was deployed into.')
+output location string = expressRoutePort.location
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type roleAssignmentType = {
```

Comment:
Both should be imported from the `avm-common-types` utl module (currently at `0.5.1`)

Reply:
both updated to import